### PR TITLE
Migrate some dev/ things to null-safety

### PR DESCRIPTION
**Pull request by goderbauer**
https://github.com/flutter/flutter/issues/71505

### CODE REVIEW

**Comment by goderbauer** [comment]
forgot to update some dependencies...

File: dev/tracing_tests/test/image_cache_tracing_test.dart [hunk 1]
```diff
@@ -12,16 +12,16 @@ import 'package:vm_service/vm_service.dart';
 import 'package:vm_service/vm_service_io.dart';
 
 void main() {
-  VmService vmService;
-  String isolateId;
+  late VmService vmService;
+  String? isolateId;
```
Comment (dnfield) [inline dev/tracing_tests/test/image_cache_tracing_test.dart:16]: not late?

**Review by dnfield — APPROVED** [review]
LGTM with nit

File: dev/tracing_tests/test/image_cache_tracing_test.dart (hunk 1 above)
Comment (goderbauer) [reply dev/tracing_tests/test/image_cache_tracing_test.dart]: Changed.

**Pull request merged**